spy.go

```diff
@@ -22,18 +22,23 @@ func ParseCommand(cmd string) (string, []string, error) {
 	return stringResult[0], stringResult[1:], nil
 }
 
-// Execute accepts a command argument and writes the result to a writer
-func Execute(w io.Writer, commandLine string) error {
+// Execute accepts two writers and a shell command
+// It writes to both writers; whilst the first writer logs only
+// the executed command result on the user's terminal
+// the second logs to a specified writer with additional messages
+func Execute(w io.Writer, w2 io.Writer, commandLine string) error {
```

Comment:
Great! Shall we give these two writers names that remind us which one is which? How about something like `terminal`, and `recorder`?